src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -621,6 +621,48 @@ public Iterable<EventBooking> findAllByEventIdAndStatus(final String eventId, @N
     }
   }
 
+  /**
+   * Find all bookings for events.
+   *
+   * @param eventIds - the event of interest.
+   * @return an iterable with all the events matching the criteria.
+   * @throws SegueDatabaseException - if an error occurs.
+   */
+  @Override
+  public Iterable<EventBooking> findAllByEventIds(final List<String> eventIds) throws SegueDatabaseException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT event_bookings.* FROM event_bookings JOIN users ON users.id=user_id WHERE event_id IN (");
+
+    // insert placeholders into the query - sb.append is adding new content to an existing string without creating
```

Comment:
Instead of having this comment, what you could do is to extract all this logic out into a private function with a descriptive name that accepts a list of event IDs and returns a `String` with the resulting SQL query.

This will make your code easier to read. For instance, later, instead of calling `conn.prepareStatement(sb.toString())` you simply call `conn.prepareStatement(sqlQuery)`